InClassAssignment2-LenaL.Rmd

```diff
@@ -0,0 +1,175 @@
+
+#Make sure that the student and teacher data frames have the same variables names
+names(V.TCH) <- c("id", "gender", "topic", "group")
+names(V.STD) <- c("id", "gender", "topic", "group")
+
+#Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
+VERTEX <- dplyr::bind_rows(V.TCH, V.STD)
+
+```
+
+Now we have both a Vertex and Edge list it is time to plot our graph!
+
+```{r}
+#Load the igraph package
+
+library(igraph)
+
+#First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+V.STD2<- dplyr::select(D1, stid, s.major)
+V.STD2<-dplyr::count(V.STD2, stid, s.major)
+V.STD2$group <- "student"
+
+
+V.TCH2<- dplyr::select(D1, tid, t.expertise)
+V.TCH2<-dplyr::count(V.TCH2, tid, t.expertise)
+V.TCH2$group <- "teacher"
+
+names(V.TCH2) <- c("id",  "topic", "count", "group")
+names(V.STD2) <- c("id",  "topic", "count","group")
+
+VERTEX2 <- dplyr::bind_rows(V.TCH2, V.STD2)
+
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+
+plot(g2,layout=layout.fruchterman.reingold, vertex.size=VERTEX2$count, edge.width=EDGE$count)
+```
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+#load the library we need
+library(dplyr)
+library(tidyr)
+ourClass<-read.csv("HUDK4050-2018-classes.csv")
+D3<-select(ourClass, Q8, Q9, Q1, Q3, Q4, Q5)
+D3<-D3 %>% unite(student,Q8,Q9,sep=' ')
+D3 <- D3[-c(1,2,39),]
+
+# formalize the data
```

Comment:
Nice work recognizing some of the formatting issues. Though you did them individually, I challenge you to think of a way to do this auto-magically, which would be useful if you wanted to rerun this code on different data.

There are a couple of ways to tackle it, but gsub should be sufficient in this case. (it's easier to remove all spaces than it is to add them.)